kafka-client/src/main/java/dev/responsive/kafka/internal/db/ColumnName.java

```diff
@@ -42,6 +43,7 @@ public enum ColumnName {
 
   static final Bytes METADATA_KEY
       = Bytes.wrap("_metadata".getBytes(StandardCharsets.UTF_8));
+  public static final long METADATA_SEGMENT_ID = -1L;
```

Comment:
I haven't gone through to verify this in the code, but just wanted to make sure that nowhere do we use the partition key `(kafkaPartition, -1L)` in atomic batches with any "data" segment? I suspect this is just used for offset/streamtime tracking?

Reply:
I think this should have been moved to the `SegmentPartitioner` -- I'll do that now. Hopefully it'll be more clear after that, but basically yes, this is only ever used for metadata rows. The only place it's used in the codebase is for the method `SegmentPartitioner#metadataTablePartition`